jaxrs/src/main/java/io/narayana/lra/filter/ServerLRAFilter.java

```diff
@@ -529,11 +530,17 @@ public void filter(ContainerRequestContext requestContext, ContainerResponseCont
 
                         progress = updateProgress(progress, ProgressStep.Ended, null);
                     }
-                } catch (NotFoundException ignore) {
-                    // must already be cancelled (if the intercepted method caused it to cancel)
-                    // or completed (if the intercepted method caused it to complete
-                    progress = updateProgress(progress, ProgressStep.Ended, null);
-                } catch (WebApplicationException | ProcessingException e) {
+                } catch (WebApplicationException e) {
+                    if (e.getResponse().getStatus() == NOT_FOUND.getStatusCode()) {
+                        // must already be cancelled (if the intercepted method caused it to cancel)
+                        // or completed (if the intercepted method caused it to complete
+                        progress = updateProgress(progress, ProgressStep.Ended, null);
```

Comment:
I am not really sure I understand this change. Before if you got a NotFoundException (which is a sub-class of WebApplicationException it seems from https://jakarta.ee/specifications/platform/9/apidocs/jakarta/ws/rs/notfoundexception) then we did updateProgress(progress, ProgressStep.Ended, null); and for other WebApplicationException we did updateProgress(progress, isCancel ? ProgressStep.CancelFailed : ProgressStep.CloseFailed, e.getMessage()); - and we seem to still be doing that but from the description of the pull request I think before we would be doing ProgressStep.Ended for either case or something?

Reply:
The original code at line 536 was catching WebApplicationException but was not checking whether the response was NOT_FOUND (which should be identical to the `catch (NotFoundException ignore)` case which is interpreted as if the LRA has already ended. I could have kept the original code on the original line 536 but changed it to test for NOT_FOUND as follows:

```
                } catch (WebApplicationException | ProcessingException e) {
                    if (e instanceof WebApplicationException &&
                       ((WebApplicationException)e).getResponse().getStatus() == NOT_FOUND.getStatusCode()) {
                        progress = updateProgress(progress, ProgressStep.Ended, null);
                    } else {
                        progress = updateProgress(progress,
                                isCancel ? ProgressStep.CancelFailed : ProgressStep.CloseFailed, e.getMessage());
                    }
```

The new code looks slightly more complex because it processed `WebApplicationException | ProcessingException` together.

Reply:
Thanks, I will resolve this thread